internal/kibana/agents.go

```diff
@@ -71,6 +72,10 @@ func (c *Client) AssignPolicyToAgent(a Agent, p Policy) error {
 
 func (c *Client) waitUntilPolicyAssigned(a Agent, p Policy) error {
 	for {
+		if signal.SIGINT() {
```

Comment:
There may be other reasons to cancel actions, and we would need to remember to add them all to all the actions that may require cancelation.

WDYT about passing these methods a context or a done channel? If it is a context it could be also passed here to `getAgent`, that could help to cancel the requests it does.

Anything that could need to handle cancelation could receive one of these contexts.

Reply:
I agree with you, but I would pair with some bigger refactoring then. Maybe a follow-up task. I think we should introduce it to all runners.

Reply:
Ok, let's revisit this later.